projects/pswg/src/main/java/com/parzivail/pswg/block/ThermalDetonatorBlock.java

```diff
@@ -122,10 +124,19 @@ public VoxelShape getOutlineShape(BlockState state, BlockView world, BlockPos po
 	@Override
 	public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit)
 	{
-		if (player.getInventory().getMainHandStack().isEmpty())
+		//if (!player.getInventory().getMainHandStack().isOf(SwgItems.Explosives.ThermalDetonator))
+		//{
+		if (player.getInventory().getMainHandStack().isOf(SwgItems.Explosives.ThermalDetonator) && state.get(CLUSTER_SIZE) < 5 && player.isSneaking())
```

Comment:
turn this `5` magic number into a named constant in the file like `public static final int MAX_CLUSTER_SIZE = 5;`